### ISSUE TITLE
capture_backtrace raises AttributeError on PEP-420 namespace packages

### ISSUE THREAD

**Issue by jacksmith15** [scoutapp/scout_apm_python#667]
The new `capture_backtrace` function in `scout_apm.core.backtrace` raises an AttributeError when the stack includes a [PEP-420] namespace package.

This is caused by the [`module_filepath` function](https://github.com/scoutapp/scout_apm_python/blob/v2.21.0/src/scout_apm/core/backtrace.py#L26-L33), specifically line 32:

```python
    module_dir = sys.modules[root_module].__file__.rsplit(os.sep, 2)[0]
```

If `sys.modules[root_module]` is a [PEP-420] namespace package, this will raise
```
AttributeError: 'NoneType' object has no attribute 'rsplit'
```

### Steps to reproduce

Create a namespace package, with some modules inside, e.g.:
```
namespace/
    foo/
        __init__.py
    bar/
       __init__.py
```

Then on an interactive Python shell:

```
>>> from scout_apm.core.backtrace import module_filepath
>>> from namespace import foo
>>> module_filepath("namespace.foo", "namespace")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jack/venvs/tmp-a17ac7185189989/lib/python3.8/site-packages/scout_apm/core/backtrace.py", line 32, in module_filepath
    module_dir = sys.modules[root_module].__file__.rsplit(os.sep, 2)[0]
AttributeError: 'NoneType' object has no attribute 'rsplit'
```

### Details

- Tested with version 2.21.0
- Current workaround is to pin version to 2.20.0

[PEP-420]: https://www.python.org/dev/peps/pep-0420/

**Comment by tim-schilling** [comment]
Hi Jack,

Thank you for the detailed issue. I'm looking into it.

-Tim

**Comment by tim-schilling** [comment]
@jacksmith15 Thank you for raising this. It will be fixed in v2.21.1 which is being released now. It'll be available within the hour.
